web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/ImportFromFileStep.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_federation\Plugin\pipeline\Step;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\joinup_federation\JoinupFederationStepPluginBase;
+use Drupal\pipeline\Exception\PipelineStepExecutionLogicException;
+use Drupal\pipeline\Plugin\PipelineStepInterface;
+use Drupal\sparql_entity_storage\Database\Driver\sparql\ConnectionInterface;
+use Drupal\sparql_entity_storage\SparqlGraphStoreTrait;
+use EasyRdf\Graph;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a step plugin that imports data from a local or remote file.
+ *
+ * @PipelineStep(
+ *   id = "file_import",
+ *   label = @Translation("Import from file"),
+ * )
+ */
+class ImportFromFileStep extends JoinupFederationStepPluginBase {
```

Comment:
So this takes either a file URI (as a local resource URI) or an external URI. Yet, it requires the devs to pre-configure the plugin with the final URL - meaning that in case of a URL change we would have to deploy a new file. Or in case of a local file update, we would still need to deploy a change because it might be that a `_1.rdf` is added to the file.
Furthermore, we have the ManualUpload step which is already doing this in a configurable way. Why not merge them both, use the file/remote_file widget we have already developed in order to make this also configurable?

Reply:
I discussed this with Claudiu and the step is intended for pipelines that run in an unmanaged mode. Thus, no interaction with the user should be needed to run this step.